### ISSUE TITLE
Crashes with 2.39 (508)

### ISSUE THREAD

**Issue by marvkis** [openziti/ziti-tunnel-apple#174]
Hi,

I have unexpected crashes with the new 2.39 (508) on 14.4. Not sure if It's related to the 14.4 update or a issue introduced in 2.39 as both updates happened nearly the same time.

It might be DNS-Related as I noticed DNS-Resolve did not response before the PacketTunnelProvider crashed, but established sessions were still working.

Some details from the crash report:
```
Process:               PacketTunnelProvider [89175]
Path:                  /Applications/Ziti Desktop Edge.app/Contents/PlugIns/PacketTunnelProvider.appex/Contents/MacOS/PacketTunnelProvider
Identifier:            io.netfoundry.ZitiPacketTunnel.PacketTunnelProvider
Version:               2.39 (508)
App Item ID:           1460484572
App External ID:       864363436
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               502

Date/Time:             2024-03-20 08:41:12.1526 +0100
OS Version:            macOS 14.4 (23E214)
Report Version:        12
Anonymous UUID:        F51B24BE-BA06-6B57-84D3-4221BAC26570

Sleep/Wake UUID:       5A411442-ED32-457F-90A9-C951968D9304

Time Awake Since Boot: 460000 seconds
Time Since Wake:       3485 seconds

System Integrity Protection: enabled

Crashed Thread:        1  Dispatch queue: com.apple.root.default-qos

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x000000049dd823e0
Exception Codes:       0x0000000000000001, 0x000000049dd823e0

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [89175]
```
```
Thread 1 Crashed::  Dispatch queue: com.apple.root.default-qos
0   PacketTunnelProvider          	       0x1026681b0 ziti_write + 84
1   PacketTunnelProvider          	       0x102612674 0x102574000 + 648820
2   PacketTunnelProvider          	       0x102618870 0x102574000 + 673904
3   PacketTunnelProvider          	       0x102632020 ip4_input_hook + 460
4   PacketTunnelProvider          	       0x102629104 ip4_input + 80
5   PacketTunnelProvider          	       0x10261a408 on_packet + 84
6   PacketTunnelProvider          	       0x1025d6fc0 0x102574000 + 405440
7   PacketTunnelProvider          	       0x1026db1a0 0x102574000 + 1470880
8   PacketTunnelProvider          	       0x1026eb8e0 0x102574000 + 1538272
9   PacketTunnelProvider          	       0x1026db6f0 uv_run + 412
10  PacketTunnelProvider          	       0x1025eb8ec 0x102574000 + 489708
11  PacketTunnelProvider          	       0x1025f1c20 0x102574000 + 515104
12  PacketTunnelProvider          	       0x1025a84e8 0x102574000 + 214248
13  libdispatch.dylib             	       0x18e35c750 _dispatch_call_block_and_release + 32
14  libdispatch.dylib             	       0x18e35e3e8 _dispatch_client_callout + 20
15  libdispatch.dylib             	       0x18e370080 _dispatch_root_queue_drain + 864
16  libdispatch.dylib             	       0x18e3706b8 _dispatch_worker_thread2 + 156
17  libsystem_pthread.dylib       	       0x18e50afd0 _pthread_wqthread + 228
18  libsystem_pthread.dylib       	       0x18e509d28 start_wqthread + 8
```

**Comment by scareything** [comment]
Hi, and thanks for the report. Based on logs that you sent separately I think this crash will be fixed by [v2.40](https://github.com/openziti/ziti-tunnel-apple/releases/tag/2.40.509), which I just pushed to test flight. Can you let me know if you see it with 2.40? Thanks.

**Comment by marvkis** [comment]
Hi. Testflight still offers 2.39-505... will have an eye on it! Thanks a lot!
